tools/kliopt/org/jetbrains/kliopt/KliOpt.kt

```diff
@@ -17,18 +17,23 @@
 
 package org.jetbrains.kliopt
 
-enum class DeprecatedState {
-    ALL {
-        override fun helpMessage(descriptor: Descriptor): String = "Warning: Option/argument -${descriptor.longName} is deprecated!"
-    },
-    ONLY_SHORT {
-        override fun helpMessage(descriptor: Descriptor): String =
-                "Warning: Short form -${(descriptor as? OptionDescriptor)?.shortName} of option -${descriptor.longName} is deprecated!"
-    },
-    ONLY_FULL {
-        override fun helpMessage(descriptor: Descriptor): String =
-                "Warning: Full form of option -${descriptor.longName} is deprecated!"
-    };
+sealed class DeprecatedState(val newName: String? = null) {
+    class AllForms(newName: String? = null) : DeprecatedState(newName) {
+        override fun helpMessage(descriptor: Descriptor) = "Warning: Option/argument -${descriptor.longName}" +
+                "${(descriptor as? OptionDescriptor)?.let { it.shortName?.let{ "-$it" } ?: ""} ?: ""} is deprecated!" +
+                "${newName?.let {" Use -$it instead."} ?: ""}"
+    }
+    class OnlyShort(newName: String? = null) : DeprecatedState(newName) {
+        override fun helpMessage(descriptor: Descriptor) =
+                "Warning: Short form -${(descriptor as? OptionDescriptor)?.shortName} of option " +
+                        "-${descriptor.longName} is deprecated!${newName?.let {" Use -$it instead."} ?: ""}"
```

Comment:
Can we advice to use `-${descriptor.longName}` instead?

Reply:
There can be cases when user wants to change old short form to new short form.